### PR TITLE
Add attendance summary field to Service Schedule

### PR DESCRIPTION
**Pull request by bmuniswamy**
# Critical Changes

# Changes
Added a the following fields on Service Schedule object

- Attendance Rate - Formula Field
- Attendance Summary - Formula Field
- Num Present Service Deliveries - Populated via a trigger or batch
- Num Absent Service Deliveries - Populated via a trigger or batch

Created a new schedulable batch for service schedule. 
Created a new feature gate record so that can be turned on when the user wants the data to be updated using a trigger. This will be turned off by default

# Issues Closed
[W-10999402](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000uxxcTYAQ/view)
# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
~~- [ ] Any net new LWC work has Sa11y tests & 50% or above lines JEST test coverage~~  
- [x] CRUD/FLS is enforced in Apex
- [x] Permission sets are updated to account for CRUD|FLS|Tab|Classes
~~- [ ] Field sets are updated to account for new fields~~
~~- [ ] UX approval or UX not necessary~~
- [x] Link the pull request and work item by PR comment and Chatter post respectively, e.g. GUS: W-0000000: Work Name (https://github.com/SalesforceFoundation/PMM/pull/303)
- [ ] All **acceptance criteria** have been met
    - [x] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [x] PR contains draft release notes
- [ ] QE story level testing completed

### CODE REVIEW

File: force-app/main/default/objects/ServiceSchedule__c/fields/AttendanceRate__c.field-meta.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>AttendanceRate__c</fullName>
```
Comment (jjbennett) [inline force-app/main/default/objects/ServiceSchedule__c/fields/AttendanceRate__c.field-meta.xml:3]: This field is not listed in the AC's do we want to verify we should add it and if yes should we then also add it to the layout?

File: force-app/main/default/classes/ServiceScheduleRollupsSchedulable.cls [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ *  * Copyright (c) 2021, salesforce.com, inc.
```
[the remark this answers is not in the document]
Comment (bmuniswamy) [reply force-app/main/default/classes/ServiceScheduleRollupsSchedulable.cls:3]: Copy and paste mistake 🤦‍♀️

File: force-app/main/default/objects/ServiceSchedule__c/fields/AttendanceRate__c.field-meta.xml (hunk 1 above)
Comment (bmuniswamy) [reply force-app/main/default/objects/ServiceSchedule__c/fields/AttendanceRate__c.field-meta.xml:3]: Also one more thing I see is that we are not displaying Attendance Rate field on the page layout for Service Session, Program Engagement, Service and Contact objects where we are calculating Attendance.

Comment (jjbennett) [reply force-app/main/default/objects/ServiceSchedule__c/fields/AttendanceRate__c.field-meta.xml:3]: Good call, probably would follow the same pattern. Let's add the AC if Meg wants does want that field as well but I assume you are probably correct.

**Review by jjbennett — APPROVED** [review]
Great work!

**Pull request merged**